docs/release_process.md

```diff
@@ -0,0 +1,269 @@
+# Release Process
+## Environments
+
+* [Production][prod] - Run by SRE team in GCP
+* [Stage][stage] - Run by SRE team in GCP
+* [Dev][dev] - Run by ENGR team in Heroku
+* Locals: Run by ENGRs on their own devices. (See [README][readme] and other [`docs/`][docs].)
+
+## Code branches
+Standard Relay development follows a branching strategy similar to
+[GitHub Flow][github-flow], where all branches stem directly from `main` and
+are merged back to `main`:
+
+1. Create a branch from `main`
+2. Make changes
+3. Create a pull request to `main`
+4. Address review comments
+5. Merge pull request
+
+```mermaid
+%%{init: { 'theme': 'base', 'gitGraph': {'rotateCommitLabel': true} } }%%
+    gitGraph
+       commit
+       branch change-1
+       commit
+       commit
+       checkout main
+       merge change-1
+```
+
+This means many features could be in development at the same time, and all can
+merge back to `main` when they are ready.
+
+```mermaid
+%%{init: { 'theme': 'base', 'gitGraph': {'rotateCommitLabel': true} } }%%
+    gitGraph
+       commit
+       branch change-1
+       commit
+       commit
+       checkout main
+       branch change-2
+       commit
+       checkout main
+       merge change-1
+       branch change-3
+       commit
+       commit
+       checkout main
+       merge change-2
+       checkout change-3
+       commit
+       commit
+       checkout main
+```
+
+## Release Timeline
+The standard release interval for Relay is 1 week, meaning every week there
+will be a new version of the Relay web app on the [Production][prod]
+environment. To do this, we first release code to [Dev][dev] and
+[Stage][stage].
+
+## Release to Dev
+~~Every commit to `main` is automatically deployed to the [Dev][dev] server.~~
+(Since the [Great GitHub Heroku Incident of 2022][github-heroku-incident], we
+have disabled GitHub integration on our Heroku app.)
+We try to push `main` to [Dev][dev] often. To do so, you need to add the heroku
+app as a remote:
+
+* `heroku git:remote -a fx-private-relay`
```

Comment:
I reflexively run `heroku login` before any `heroku` commands - is that needed in the docs?

Reply:
Good idea.